### ISSUE TITLE
[bug #9882] faulty flash writes on ATmega32 via avr910

### ISSUE THREAD

**Issue by avrdude-admin** [avrdudes/avrdude#20]
none given <none given>
Wed 04 Aug 2004 02:44:43 PM UTC

I'm getting very bizarre faults on writing flash, looks like either buggy avr910 driver code or timing problems - see the attached file for a detailed description.  Hardware works fine with AVRprog.exe, host and target work fine with avrdude and an avrisp programmer, so the bug must be specific to avrdude's avr910 code.

[file #1630: problem.txt](https://savannah.nongnu.org/bugs/download.php?file_id=1630)

This issue was migrated from https://savannah.nongnu.org/bugs/?9882

**Comment by avrdude-admin** [comment]
Tue 10 Aug 2004 05:30:37 PM UTC

Okay, the problems I experienced on non-interactive downloads were due to failure in avr910_paged_write_flash() to re-set the address correctly after page writes.  That function has a several other problems:

- no polling or delay after page writes;
- no page writes when not using auto-increment;
- an extraneous page write when data ends on page boundary.

I'll submit a patch fixing all of the above.

The mess I observed in interactive mode is another story: avr910_write_byte() simply doesn't know at all how to handle paged memory.  Ugh.

- nic
